auth.php

```diff
@@ -167,8 +169,14 @@ public function logOff() {
     * @return  bool             true on successful auth
     */
     public function trustExternal($user, $pass, $sticky = false) {
-        if(!is_null($this->chained_auth) && $this->chained_auth->canDo('external'))
-            $this->chained_auth->trustExternal($user, $pass, $sticky);
+        foreach($this->chained_plugins as $module) {
+            if($module[1]->canDo('external') && $module[1]->trustExternal($user, $pass)) {
+                $_SESSION[DOKU_COOKIE]['plugin']['authchained']['module'] = $module[0];
+                $this->chained_auth = $module[1];
+                return true;
+            }
+        }
+        return false;
```

Comment:
Here you'd need to trigger the "auth_login" event (same as "auth_setup" from auth.php does after checking wether a plugin can do external auth) instead of just returning false. You can learn how to from auth.php - see [Xref for auth.php](https://xref.dokuwiki.org/reference/dokuwiki/nav.html?inc/auth.php.source.html#l99): 'else', lines 103-109.